src/server/misc/cmds/cmds.go

```diff
@@ -175,6 +185,85 @@ func Cmds(ctx context.Context) []*cobra.Command {
 	}
 	commands = append(commands, cmdutil.CreateAlias(decodeURL, "misc decode-download-url"))
 
+	testMigrations := &cobra.Command{
+		Use:   "{{alias}} <postgres dsn>",
+		Short: "Runs the database migrations against the supplied database, then rolls them back.",
+		Long:  "Runs the database migrations against the supplied database, then rolls them back.",
+		Run: cmdutil.RunFixedArgs(1, func(args []string) (retErr error) {
+			ctx, c := signal.NotifyContext(pctx.Background(""), os.Interrupt)
+			defer c()
+
+			dsn := args[0]
+			db, err := sqlx.Open("pgx", dsn)
+			if err != nil {
+				return errors.Wrap(err, "open database")
+			}
+			if err := dbutil.WaitUntilReady(ctx, db); err != nil {
+				return errors.Wrap(err, "wait for database ready")
+			}
+
+			// Create test dirs for etcd data
+			dir, err := os.MkdirTemp("", "test-migrations")
+			if err != nil {
+				return errors.Wrap(err, "create etcd server tmpdir")
+			}
+			defer os.RemoveAll(dir)
+
+			etcdConfig := embed.NewConfig()
+			etcdConfig.MaxTxnOps = 10000
+			etcdConfig.Dir = filepath.Join(dir, "dir")
+			etcdConfig.WalDir = filepath.Join(dir, "wal")
+			etcdConfig.InitialElectionTickAdvance = false
+			etcdConfig.TickMs = 10
+			etcdConfig.ElectionMs = 50
+			etcdConfig.ListenPeerUrls = []url.URL{}
+			etcdConfig.ListenClientUrls = []url.URL{{
+				Scheme: "http",
+				Host:   "localhost:7777",
```

Comment:
We should use :0 but we can't read it back out of the etcd client.  The tests work around this by creating a listener on port :0, seeing what port it picked, and then using that port explicitly, but that doesn't really work because it's a race condition between when we destroy the :0 listener and when we start listening on the selected port.  I didn't bother to duplicate this; if they are running something on port 7777, kill that first.  It's pretty unlikely that someone runs into this, unless they accidentally run two copies of this script at once, which would fail anyway.  It's true that they could have picked 7777 for their favorite service, but it's not officially allocated or particularly common, and they can run always run this in a network namespace if they can't kill what's on port 7777 for 5 minutes.

This fake etcd only has to live long enough to not cause errors in migrations that read from etcd (there is a move enterprise stuff from etcd to postgres that requires it), so nobody will need to connect to this.

It might be worth making this script connect to etcd, but it isn't relevant for 2.5 -> 2.6 upgrades.